File: src/vb/consoleUtil.vb
Imports System

Public Class ConsoleUtil
    Public Shared Function Prompt(ByVal question As String)
        Console.WriteLine(question)
        Prompt = Console.ReadLine()
    End Function

    Public Shared Sub DisplayMatrixResult(ByVal result As Results.MatrixResult)
        Dim matrix = result.Matrix
        Dim length = Math.Max((result.Side * result.Side).ToString().Length + 1, 2)

        For rowIndex = 0 To result.Side - 1
            Dim rowText = ""
            For columnIndex = 0 To result.Side -1
                rowText += matrix(rowIndex, columnIndex).ToString().PadLeft(length, " ")
            Next columnIndex
            Console.WriteLine(rowText)
        Next rowIndex

        Console.Write($"Sum: {result.Sum}")
    End Sub
End Class

File: src/cs/consoleUtil.cs
using System;
using static System.Console;

public static class ConsoleUtil 
{
    public static string? Prompt(string question)
    {
        WriteLine(question);
        return ReadLine();
    }

    public static void DisplayMatrixResult(Results.MatrixResult  result)
    {
        var matrix = result.Matrix;
        var length = Math.Max(Math.Pow(result.Side, 2).ToString().Length + 1, 2);

        for (var rowIndex = 0; rowIndex < result.Side; rowIndex++) {
            var rowText = "";
            for (var columnIndex = 0; columnIndex < result.Side; columnIndex++) {
                rowText += matrix[rowIndex, columnIndex].ToString().PadLeft(length, ' ');
            }
            WriteLine(rowText);
        }

        WriteLine("Sum: " + result.Sum);
    }
}
